app/controllers/dashboard/base_controller.rb

```diff
@@ -0,0 +1,15 @@
+class Dashboard::BaseController < ApplicationController
+  before_action :require_login
+  rescue_from ActiveRecord::RecordNotFound, with: :not_found
+
+  private
+
+  def not_authenticated
+    redirect_to login_path, alert: t(:please_log_in)
+  end
+
+  def not_found
+    flash[:alert] = 'Вы обратились к несуществующей записи.'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.